lib/Dialect/FIRRTL/Transforms/RemoveInvalid.cpp

```diff
@@ -121,10 +125,29 @@ void RemoveResetsPass::runOnOperation() {
     }
   }
 
+  for (auto inv : llvm::make_early_inc_range(
```

Comment:
Can you build a worklist of invalid values in the first loop?  It seems likely cheaper than two walks.

Reply:
Yes. (I noticed this, but was trying to keep the initial patch obvious.) I'll add a follow-up that converts the first walk to compute a worklist of invalids to visit.